### PR TITLE
Add `ReduceScatter` XLA collective

### PR DESCRIPTION
**Pull request by jysohn23**

### CODE REVIEW

**Comment by jysohn23** [comment]
@JackCaoG bump..!

**Comment by JackCaoG** [comment]
> @JackCaoG bump..!

Sorry for the delay! Looking...

File: test/test_mp_reduce_scatter.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import sys
+import torch
+import torch_xla.core.xla_model as xm
+import torch_xla.distributed.xla_multiprocessing as xmp
+
+
+def _mp_fn(index):
+  device = xm.xla_device()
+  world_size = xm.xrt_world_size()
+  scale = 1.0
```
Comment (JackCaoG) [inline test/test_mp_reduce_scatter.py:10]: maybe it would be better to test the scale that is not 1.0?

**Review by JackCaoG — APPROVED** [review]
Thanks @jysohn23 ! minor nit

File: test/test_mp_reduce_scatter.py (hunk 1 above)
Comment (jysohn23) [reply test/test_mp_reduce_scatter.py]: Thanks, done.

**Review by JackCaoG — APPROVED** [review]

**Pull request merged**